stdlib/public/core/LifetimeManager.swift

```diff
@@ -38,6 +40,7 @@ public func withExtendedLifetime<T, Result>(
 /// - Returns: The return value, if any, of the `body` closure parameter.
 @inlinable
 public func withExtendedLifetime<T, Result>(
+  // FIXME(NCG): This should have T, Result as ~Copyable, but then the closure would need to take a borrow
```

Comment:
It seems adding `borrowing` to the closure parameter does not change the default convention for this thing thus it doesn't affect the mangling as well. It should be possible to say:
```swift
public func withExtendedLifetime<T: ~Copyable, Result: ~Copyable>(
  _ x: borrowing T,
 _ body: (borrowing T) throws -> Result
) rethrows -> Result 
```
Alternatively, we can make this interface `@usableFromInline internal` and make the noncopyable version AEIC that also supports typed throws.

Reply:
Changing the function type's argument to `borrowing` would be source breaking, as it disables implicit copying in closures that get passed to `withExtendedLifetime`. 

I think this behavior is good, as we do want `borrowing` to give us better control over copying! We can probably live with API additions that opt into this.